hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/SchedulerUtils.java

```diff
@@ -403,7 +407,8 @@ public static boolean checkQueueLabelExpression(Set<String> queueLabels,
         // check node label manager contains this label
         if (null != rmContext) {
           RMNodeLabelsManager nlm = rmContext.getNodeLabelManager();
-          if (nlm != null && !nlm.containsNodeLabel(str)) {
+          if (nlm != null && !nlm.containsNodeLabel(str)
+              && !str.equals(RMNodeLabelsManager.ANY)) {
```

Comment:
I can't understand the javadoc for this method. Could you please fix that and add how you have modified the behavior?